qa/acceptance-tests/src/test/java/io/camunda/it/cluster/dynamicnodeid/HealthCheckIT.java

```diff
@@ -0,0 +1,173 @@
+                      .withAdditionalProperties(
+                          Map.of(
+                              "camunda.cluster.dynamic-node-id.type",
+                              "s3",
+                              TASK_ID_PROPERTY,
+                              UUID.randomUUID().toString(),
+                              "camunda.cluster.dynamic-node-id.s3.bucketName",
+                              BUCKET_NAME,
+                              "camunda.cluster.dynamic-node-id.s3.leaseDuration",
+                              LEASE_DURATION,
+                              "camunda.cluster.dynamic-node-id.s3.endpoint",
+                              toxiproxyS3Endpoint,
+                              "camunda.cluster.dynamic-node-id.s3.region",
+                              S3.getRegion(),
+                              "camunda.cluster.dynamic-node-id.s3.accessKey",
+                              S3.getAccessKey(),
+                              "camunda.cluster.dynamic-node-id.s3.secretKey",
+                              S3.getSecretKey(),
+                              "camunda.cluster.dynamic-node-id.s3.apiCallTimeout",
+                              LEASE_DURATION.dividedBy(2).toString())))
+          .build();
+
+  @BeforeAll
+  public static void setupAll() throws IOException, URISyntaxException {
+    s3Client =
+        S3NodeIdRepository.buildClient(
+            new S3ClientConfig(
+                Optional.of(new S3ClientConfig.Credentials(S3.getAccessKey(), S3.getSecretKey())),
+                Optional.of(Region.of(S3.getRegion())),
+                Optional.of(S3.getEndpoint())));
+
+    // bucket must be created before the application is started
+    s3Client.createBucket(b -> b.bucket(BUCKET_NAME));
+
+    proxy = PROXY_REGISTRY.getOrCreateHostProxy(S3.getEndpoint().getPort());
+
+    // Get the Toxiproxy endpoint
+    final var s3Endpoint = S3.getEndpoint();
+    toxiproxyS3Endpoint =
+        new URI(
+            s3Endpoint.getScheme(),
+            s3Endpoint.getUserInfo(),
+            "0.0.0.0",
+            TOXIPROXY.getMappedPort(proxy.internalPort()),
+            s3Endpoint.getPath(),
+            s3Endpoint.getPath(),
+            s3Endpoint.getFragment());
+  }
+
+  @Test
+  public void shouldNotReturnHealthyWhenS3IsNotAvailable() throws IOException {
+    final var actuators =
+        testCluster.brokers().values().stream()
+            .collect(Collectors.toMap(TestStandaloneBroker::nodeId, BrokerHealthActuator::of));
+    assertThat(actuators.values())
+        .allSatisfy(actuator -> assertThatNoException().isThrownBy(actuator::ready));
+    // When:
+    // - toxiproxy introduces 10s latency in the HTTP calls
+    // - broker tries to check health & s3 is not able to reply in time
+    proxy.proxy().toxics().latency("latency", ToxicDirection.UPSTREAM, 10000L);
+    // Then: health check should fail
+    // check is run in parallel because the repository is timing out
+    final var futures =
```

Comment:
Is there a possibility of flakiness here? What happens if the actuator call takes longer than 10 seconds?